### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by jacquelinekay**
initial implementation of the header laid out in the master branch.

Some highlights:

See "examples/example_loop" for a common usage of this library.

Argument parsing and initial memory allocation is taken care in `rttest_read_args`.

The code to be instrumented is passed as a function pointer to `rttest_spin`, which calls the function and measures scheduling jitter and pagefaults in a real-time safe way.

Teardown happens in `rttest_finish`.

`rttest` keeps track of statistics and stores parameters for different threads, but the interface to rttest does not expose the thread. It keeps an internal struct to check the thread id of the calling thread and gets/sets data for that thread accordingly.

### CODE REVIEW

**Comment by wjwwood** [comment]
The approach makes sense to me, though I didn't take the time to pick the code apart for logic errors.

File: include/rttest/rttest.h [hunk 1]
```diff
@@ -17,127 +17,137 @@
...
+
+/// \brief Create a new rttest instance for a new thread.
+/// The thread's parameters are based on the first thread that called rttest_init.
+/// To be called directly after the user creates the thread.
+int rttest_init_new_thread();
+
+/// \brief Spin at the specified wakeup period for the specified number of
+/// iterations.
+/// \param[in] user_function Function pointer to execute on wakeup
+/// \param[out] Error code to propagate to main function.
```
Comment (tfoote) [inline include/rttest/rttest.h]: copy paste error on args

**Comment by tfoote** [comment]
It looks like several of the docstrings in rttest.h are out of sync with the prototypes. Otherwise lgtm

**Comment by jacquelinekay** [comment]
I update the docstrings and removed a few unused input parameters in ff8f0d5e0fc67d4cf57f9fc8010ef96cd48a4156

**Comment by jacquelinekay** [comment]
I've updated the CMakeLists to look for `ament_cmake` and build rttest as an ament package if it was found.

**Pull request merged**